### PR TITLE
OpenSearch user permissions documented

### PR DESCRIPTION
**Pull request by jashaik**
Signed-off-by: jan shahid shaik <jashaik@progress.com>

### Description

[Please describe what this change achieves]
Documented the Opensearch minimum permissions for all the index operations
https://github.com/chef/customer-bugs/issues/717

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]
In this PR
### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: docs-chef-io/content/server/upgrades.md [hunk 1]
```diff
@@ -60,6 +60,12 @@ Chef Infra Server 14.14 supports external OpenSearch for indexing. Please follow
 1. Set the `opscode_erchef['search_auth_username']` attribute to OpenSearch username.
 1. Set the `opscode_erchef['search_auth_password']` attribute to OpenSearch password.
 
+{{< note >}}
+
+The OpenSearch user should have full access to the cluster: all cluster-wide operations, write to all indices. We recommend the backend role as admin `backend_roles:- "admin"`. Please refer the OpenSearch Roles for more details: https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#predefined-roles
```
Comment (PrajaktaPurohit) [inline docs-chef-io/content/server/upgrades.md:65]: Would the admin user have the all_access role? I do not see admin user listed in the documentation provided.

Comment (jashaik) [reply docs-chef-io/content/server/upgrades.md:65]: Yes Prajakta. Here is the link for the admin role details - https://opensearch.org/docs/latest/security-plugin/configuration/yaml#roles_mappingyml

Comment (PrajaktaPurohit) [reply docs-chef-io/content/server/upgrades.md:65]: Ahh - I see, perhaps we can also include this link in the docs. 
Couple of questions:
- Could this note be after the example below?
- Is this opensearch user the same as the `opscode_erchef['search_auth_username']` that we would set on server?
- Could we clarify that we expect this user to be created by whoever is setting up external elasticsearch? (i.e reconfigure or any setup commands for server will not create this user on external opensearch setups)

Comment (jashaik) [reply docs-chef-io/content/server/upgrades.md:65]: This user is same user as opscode_erchef['search_auth_username']. I will update the docs accordingly.
Thanks for the valuable feedback.

Comment (jashaik) [reply docs-chef-io/content/server/upgrades.md]: Done

File: docs-chef-io/content/server/upgrades.md [hunk 2]
```diff
@@ -72,6 +72,13 @@ opscode_erchef['search_auth_username'] = "OPEN_SEARCH_USER"
 opscode_erchef['search_auth_password'] = "OPEN_SEARCH_PWD"
 ```
 
+{{< note >}}
+
+The OpenSearch user should have full access to the cluster: all cluster-wide operations, write to all indices. We recommend the backend role as admin `backend_roles:- "admin"`. Please refer the OpenSearch Roles for more details: https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#predefined-roles and https://opensearch.org/docs/latest/security-plugin/configuration/yaml#roles_mappingyml.
+
+We expect this user to be created by whoever is setting up external elasticsearch.`chef-server-ctl reconfigure` or setup commands for server will not create this user on external opensearch setups.
```
Comment (IanMadd) [inline docs-chef-io/content/server/upgrades.md:79]:
```suggestion
The OpenSearch user should have full access to the cluster, including access to all cluster-wide operations and the ability to write to all indices. We recommend that the user has the admin backend role. 

Please refer to OpenSearch's documentation on [predefined roles](https://opensearch.org/docs/latest/security-plugin/access-control/users-roles/#predefined-roles) and [role mapping configuration](https://opensearch.org/docs/latest/security-plugin/configuration/yaml#roles_mappingyml).

This user must be created on the external OpenSearch cluster. The Chef Infra Server executable cannot be used to create this user on external OpenSearch setups.
```

**Pull request merged**